scripts/docsgen-cli/flag_formatter.go

```diff
@@ -0,0 +1,76 @@
+package main
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/urfave/cli/v2"
+)
+
+// FlagFormatter handles flag formatting
+type FlagFormatter struct {
+	flag cli.Flag
+}
+
+// NewFlagFormatter creates a new flag formatter
+func NewFlagFormatter(flag cli.Flag) *FlagFormatter {
+	return &FlagFormatter{flag: flag}
+}
+
+// Format formats the flag string with optional environment variables
+func (ff *FlagFormatter) Format() string {
+	names, usage, value, envVars := ff.GetData()
+
+	var formatted string
+	if value != "" {
+		formatted = fmt.Sprintf("   %-30s %s (default: %s)", names, usage, value)
```

Comment:
```suggestion
		defpfx := " "
		if usage == "" {
			defpfx = ""
		}
		formatted = fmt.Sprintf("   %-30s %s%s(default: %s)", names, usage, defpfx, value)
```

something like this would deal with the case where there's no usage string; alternatively you could extend `usage` by one space if it's not `""`.
I'm not sure this is strictly needed when you account for the `Hidden` flags but it would fix current alignment problems.

Reply:
Also this `-30` is getting in the way when the `names` is too long. Like when `--miner-repo value, --storagerepo` comes in to play which is longer than you want.

Here's what I think I'd do given your current code structure so as not to have to make too many changes: make the this number a property of `FlagFormatter` that you pass in when you construct it (or make a default value you can override if you like) (you'll probably need to `fmt.Sprintf` the format itself to insert it). Then you do two passes iterating through your flags - the first pass looks at the maximum length of the `names` from `GetData` and then the second pass sets the maximum length if it's greater than 30 and does the actual print.